google/resource_compute_instance.go

```diff
@@ -488,6 +488,33 @@ func resourceComputeInstance() *schema.Resource {
 				},
 			},
 
+			"shielded_vm_config": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enable_secure_boot": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
```

Comment:
Maybe it would be better to default to true?

Reply:
Sticking to the API default- I'd imagine `false`- is our best option here imo.

Reply:
Did some more testing, and it looks like the api defaults are actually
```
secureBoot - false
enableVtpm - true
enableIntegrityMonitoring - true
```

This feels confusing to me and my instinct is to stick with `false` but let me know if that doesn't seem right to you.

Reply:
I'm inclined to say we should mirror the API defaults; Terraform only computes default values if a block is defined, so we'd see different values when we defined

```hcl
shielded_vm_config {}
```

vs not defining a block at all.

Reply:
that's a really good point